datafusion/physical-expr/src/expressions/binary.rs

```diff
@@ -1689,7 +1874,524 @@ pub fn interval_scalar_interval_op(
             scalar.get_datatype(),
         )))?,
     };
-    Ok(ColumnarValue::Array(ret))
+    Ok(ret)
+}
+
+/// This function handles interval +/- interval operations where the former is
+/// a scalar and the latter is an array, resulting in an interval array.
+pub fn scalar_interval_op_interval(
+    scalar: &ScalarValue,
+    sign: i32,
+    array: &ArrayRef,
+) -> Result<ArrayRef> {
+    let ret = match (scalar, array.data_type()) {
```

Comment:
same with above.
We can use use ScalarValue::* use DataType::* use IntervalUnit::* to simplify code